tools/README.md

```diff
@@ -49,84 +49,68 @@ This is a small tool to manipulate with secret key. Along with generating new on
 
-Command stands for data to be derived from the base secret key. Commands have options (not mandatory), which vary how data will be displayed.
+Each command stands for data to be derived from the base secret key. For the description and list of arguments for each command execute `dscp-keygen <command> --help`.
 
-List of options types:
-
-* View - determines how to display raw data, can be `raw`, `base64` or `hex`.
-* Prettiness - for large outputs, whether to use pretty multiline output.
-  Can be `pretty` or `one-line`
-* Password - password used to generate secret key.
-
-Following commands are supported:
-
-* `secret[:<view=base64>]` - display secret itself.
-* `public[:<view=hex>]` - display public key.
-* `address` - address, base58bitcoin.
-* `esecret[:<password="">[:<view=hex>]]` - display encrypted secret key.
-* `keyfile[:<password="">[:<prettiness=pretty>]]` - display keyfile.
-* `educator-auth:endpointName` - produce JWT token for educator node from given educator secret key and name of accessed endpoint.
-* `student-submission[:<seed="">]` - produce JSON request for `POST /api/student/v1/submissions` endpoint.
+Note, that name of command argument sometimes collide with options related to the input (e.g. `--password` may relate to secret on input or to secret on output).
+In such cases, all options before the command name are considered related to the input, and options after the command name - to the output.
 
 ### Examples
 
-_Reminder: no option after `:` is mandatory._
-
 * Generate secret key:
 
 ```bash
-echo 1000 | dscp-keygen --seed --command secret
+echo 1000 | dscp-keygen --seed secret
 ```
 
 * Generate truly random secret:
 
 ```bash
-cat /dev/urandom | head -c 32 | dscp-keygen --seed --command secret
+cat /dev/urandom | head -c 32 | dscp-keygen --seed secret
 ```
 
 * Generate key file with password:
 
 ```bash
-echo 1000 | dscp-keygen --seed --command keyfile:password > secret.key
+echo 1000 | dscp-keygen --seed keyfile --password=qwerty123 > secret.key
 chmod 660 secret.key
 ```
 
 * Read key file:
 
 ```base
-cat secret.key | dscp-keygen --keyfile --password qwerty123 --command secret
+cat secret.key | dscp-keygen --keyfile --password qwerty123 secret
 ```
 
 * Produce public key:
 
 ```base
-echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret  --command "public:hex"
+echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret public --hex
 ```
 
 * Produce address:
 
 ```base
-echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret --command "address"
+echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret address
 ```
 
 * Encrypt secret key:
 
 ```base
-echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret --command "esecret:12345678"
+echo '9JsCkzFH/W7SbgvgokeJTsckJs/uoAPWWwC40Y6UfFo=' | dscp-keygen --secret esecret --password=12345678
```

Comment:
Typo.

Reply:
Do you mean `esecret`? It is a command for printing an encrypted secret.